src/appengine/handlers/cron/grouper.py

```diff
@@ -219,8 +219,8 @@ def group_testcases():
 
 def group_testcases_with_same_issues(testcase_map):
   """Group testcases that are associated with same underlying issue."""
-  for testcase_1_id, testcase_1 in testcase_map.iteritems():
-    for testcase_2_id, testcase_2 in testcase_map.iteritems():
+  for testcase_1_id, testcase_1 in list(testcase_map.items()):
```

Comment:
I'm a little surprised by these changes, since the previous version was using generators and this change makes it always convert the whole thing to a list. I would've expected the change to omit the list() call so that under Python 3 the behaviour is the same as before.

Reply:
https://stackoverflow.com/questions/17695456/why-does-python-3-need-dict-items-to-be-wrapped-with-list :( :( i can remove all the list() since it was just overcautionary but we probably have to rerun futurize sometime and need to find way so that it does not reconvert it again.

Reply:
Probably will convert to option 2 in https://python-future.org/compatible_idioms.html#iterating-through-dict-keys-values-items

```
# Python 2 and 3: option 3
from future.utils import iteritems
# or
from six import iteritems

for (key, value) in iteritems(heights):
    ...
```

Reply:
Using six sounds good! Converting everything to a list could potentially have issues because of extra memory etc.